src/reducer/filterReducer.js

```diff
@@ -0,0 +1,159 @@
+      return { ...state, succulentPlants: !state.succulentPlants };
+
+    case "FLOWERING_PLANTS":
+      return { ...state, floweringPlants: !state.floweringPlants };
+
+    case "EVERGREEN_PLANTS":
+      return { ...state, evergreenPlants: !state.evergreenPlants };
+
+    case "ONE_RATING_PLUS":
+      return { ...state, oneRatingPlus: state.oneRatingPlus == 0 ? 1 : 0 };
+
+    case "TWO_RATING_PLUS":
+      return { ...state, twoRatingPlus: state.twoRatingPlus == 0 ? 2 : 0 };
+
+    case "THREE_RATING_PLUS":
+      return {
+        ...state,
+        threeRatingPlus: state.threeRatingPlus == 0 ? 3 : 0,
+      };
+
+    case "FOUR_RATING_PLUS":
+      return { ...state, fourRatingPlus: state.fourRatingPlus == 0 ? 4 : 0 };
+
+    case "CLEAR_FILTERS":
+      return initialFilterState;
+
+    default:
+      return state;
+  }
+}
+
+function applyFilters(state, products) {
+  let filterProducts = products;
+
+  if (state.priceLowToHigh) {
+    filterProducts = filterProducts.sort((a, b) => a.price - b.price);
+  }
+
+  if (state.priceHighToLow) {
+    filterProducts = filterProducts.sort((a, b) => b.price - a.price);
+  }
+
+  if (state.priceRange) {
+    filterProducts = filterProducts.filter(
+      ({ price }) => Number(price) <= state.priceRange
+    );
+  }
+
+  if (state.indoorPlants) {
+    filterProducts = filterProducts.filter(
+      ({ categoryName }) => categoryName.toUpperCase() == "INDOOR PLANT"
+    );
+  }
+
+  if (state.outdoorPlants) {
+    filterProducts = filterProducts.filter(
+      ({ categoryName }) => categoryName.toUpperCase() == "OUTDOOR PLANT"
+    );
+  }
+
+  if (state.officePlants) {
+    filterProducts = filterProducts.filter(
+      ({ categoryName }) => categoryName.toUpperCase() == "OFFICE PLANT"
+    );
+  }
+
+  if (state.succulentPlants) {
+    filterProducts = filterProducts.filter(
+      ({ plantType }) => plantType.toUpperCase() == "SUCCULENT"
+    );
+  }
+
+  if (state.evergreenPlants) {
+    filterProducts = filterProducts.filter(({ plantType }) =>
+      plantType.toUpperCase().includes("EVERGREEN")
+    );
+  }
+
+  if (state.floweringPlants) {
+    filterProducts = filterProducts.filter(({ plantType }) =>
+      plantType.toUpperCase().includes("FLOWERING")
+    );
+  }
+
+  if (state.oneRatingPlus) {
+    filterProducts = filterProducts.filter(
+      ({ rating }) => rating >= state.oneRatingPlus
+    );
+  } else if (state.twoRatingPlus) {
+    filterProducts = filterProducts.filter(
+      ({ rating }) => rating >= state.twoRatingPlus
+    );
+  } else if (state.threeRatingPlus) {
+    filterProducts = filterProducts.filter(
+      ({ rating }) => rating >= state.threeRatingPlus
+    );
+  } else if (state.fourRatingPlus) {
+    filterProducts = filterProducts.filter(
+      ({ rating }) => rating >= state.fourRatingPlus
+    );
+  }
+
+  return filterProducts;
+}
+
+const useFilter = () => {
+  const [state, dispatch] = useReducer(filterReducer, initialFilterState);
```

Comment:
nitpicking: you can give more meaningful names to these like filters and filtersDispatch.

Reply:
Sure ,I will do that, thanks for the review